config/src/sumeragi.rs

```diff
@@ -94,11 +94,16 @@ impl ConfigurationProxy {
     pub fn insert_self_as_trusted_peers(&mut self) {
         let peer_id = self
             .peer_id
-            .clone()
+            .as_ref()
             .expect("Insertion of `self` as `trusted_peers` implies that `peer_id` field should be initialized");
-        self.trusted_peers = Some(TrustedPeers {
-            peers: unique_vec![peer_id],
-        });
+        self.trusted_peers = if let Some(mut trusted_peers) = self.trusted_peers.take() {
+            trusted_peers.peers.push(peer_id.clone());
```

Comment:
What if self already present in `trusted_peers`?

Reply:
no problem, it's backed by `unique_vec` so it will not add a duplicate

Reply:
oh indeed